src/Controller/Api/BookController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+namespace App\Controller\Api;
+
+use App\Entity\Book;
+use App\Service\BookManager;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+
+/**
+ * Class BookController
+ * @package App\Controller\Api
+ *
+ */
+class BookController extends AbstractFOSRestController
+{
+    private BookManager $bookManager;
+
+    public function __construct( BookManager $bookManager)
+    {
+        $this->bookManager = $bookManager;
+    }
+
+    /**
+     * @Rest\Get(path="/api/books")
+     * @Rest\View(serializerGroups={"Default"})
+     * @QueryParam(name="page", requirements="\d+", default="1", description="The page number")
+     * @QueryParam(name="name", default="", description="The name of the book")
+     * @QueryParam(name="genre", default="", description="The genre of the book")
+     *
+     * @param ParamFetcherInterface $paramFetcher
+     * @return View
+     */
+    public function getBooks(ParamFetcherInterface $paramFetcher): View
```

Comment:
If you're using something to extract parameters for your query, it should happen either before the control is given to the method, or inside the method. In any case, a controller method must receive a determined list of dependencies to work with. Now, it is a black box.